src/helpers/daily-data-providers/combined-resting-heart-rate.ts

```diff
@@ -1,42 +1,73 @@
+    endDate: Date,
+): Promise<Record<string, number>> {
+    const providers: Promise<Record<string, number>>[] = [];
 
-	return MyDataHelps.getDataCollectionSettings().then((settings) => {
-		if (settings.fitbitEnabled) {
-			providers.push(fitbitRestingHeartRateDataProvider(startDate, endDate));
-		}
-		if (settings.garminEnabled) {
-			providers.push(garminRestingHeartRateDataProvider(startDate, endDate));
-		}
-		if (settings.queryableDeviceDataTypes.find(s => s.namespace == "AppleHealth" && s.type == "RestingHeartRate")) {
-			providers.push(appleHealthRestingHeartRateDataProvider(startDate, endDate));
-		}
+    const combinedSettings = await getCombinedDataCollectionSettings();
+    const { settings, deviceDataV2Types } = combinedSettings;
 
-		if (!providers.length) {
-			return {};
-		}
+    if (settings.fitbitEnabled) {
+        providers.push(fitbitRestingHeartRateDataProvider(startDate, endDate));
+    }
+    if (settings.garminEnabled) {
+        providers.push(garminRestingHeartRateDataProvider(startDate, endDate));
+    }
+    if (
+        settings.appleHealthEnabled &&
+        settings.queryableDeviceDataTypes.some(
+            (ddt) =>
+                ddt.namespace === "AppleHealth" &&
+                ddt.type === "RestingHeartRate",
+        )
+    ) {
+        providers.push(
+            appleHealthRestingHeartRateDataProvider(startDate, endDate),
+        );
+    }
+    if (
+        settings.healthConnectEnabled &&
+        deviceDataV2Types.some(
+            (ddt) =>
+                ddt.namespace === "HealthConnect" &&
+                ddt.type === "resting-heart-rate",
+        )
+    ) {
+        providers.push(
+            healthConnectRestingHeartRateDataProvider(startDate, endDate),
+        );
+    }
 
-		return Promise.all(providers).then((values) => {
-			var data: { [key: string]: number } = {};
-			while (startDate < endDate) {
-				var dayKey = getDayKey(startDate);
-				var heartRates: number[] = [];
-				values.forEach((value) => {
-					if (value[dayKey] && value[dayKey] > 0) {
-						heartRates.push(value[dayKey]);
-					}
-				});
-				if (heartRates.length > 0) {
-					data[dayKey] = Math.round( heartRates.reduce((a,b) => a+b) / heartRates.length );
-				}
-				startDate = add(startDate, { days: 1 });
-			}
-			return data;
-		});
-	});
+    if (!providers.length) {
+        return {};
+    }
+
+    const values = await Promise.all(providers);
+    const data: Record<string, number> = {};
+    while (startDate < endDate) {
+        let dayKey = getDayKey(startDate);
+        let heartRates: number[] = [];
+        values.forEach((value) => {
+            if (value[dayKey] && value[dayKey] > 0) {
+                heartRates.push(value[dayKey]);
+            }
+        });
+        if (heartRates.length > 0) {
+            data[dayKey] = Math.round(
+                heartRates.reduce((a, b) => a + b) / heartRates.length,
+            );
+        }
+        startDate = add(startDate, { days: 1 });
```

Comment:
I know this was pre-existing, but I'd suggest using an intermediate variable to store the current date value for the loop.  I think it is generally considered bad practice to modify the function parameter reference.  By changing the referenced value here, it no longer actually represents the start date, so it could lead to confusion or possibly coding errors for future maintainers (e.g. if someone doesn't notice that `startDate` was modified and tries to use it after the for loop, expecting the original value).

Not a big deal, but as long as we are adjusting this, might as well clean that up, too.